File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string w;
int answer = 0, cnt = 0;
char alp[5] = {'A', 'E', 'I', 'O', 'U'};

void search(string s) {
    //cout << s << "(" << s.length() << ")" << " : " << cnt << endl;
    if (s == w) {
        answer = cnt;
        return;
    }
    if (s.length() == 5) {
        return;
    }
    
    for (int i = 0; i < 5; i++) {
        cnt += 1;
        search(s + alp[i]);  
    }
}

int solution(string word) {
    w = word;
    
    search("");
    
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
import sys
sys.setrecursionlimit(10000)

letters = ['A', 'E', 'I', 'O', 'U']

answer = 0
count = 0
def dfs(start, word):
    global answer, count
    
    # 종료 조건
    if start == word:
        answer = count
        return 
    
    if len(start) < 5:
        for l in letters:
            count += 1
            dfs(start + l, word)

def solution(word):
    dfs('', word)       
    global answer
    
    return answer